lib/ansible/plugins/strategy/__init__.py

```diff
@@ -315,6 +320,27 @@ def _queue_task(self, host, task, task_vars, play_context):
             queued = False
             starting_worker = self._cur_worker
             while True:
+                # Determine the "rewind point" of the worker list. This means we start
+                # iterating over the list of workers until the end of the list is found.
+                # Normally, that is simply the length of the workers list (as determined
+                # by the forks or serial setting), however a task/block/play may "throttle"
+                # that limit down.
+                templar = Templar(loader=self._loader, variables=task_vars)
+                try:
+                    throttle = int(templar.template(task.throttle))
+                except Exception as e:
+                    raise AnsibleError("Failed to convert the throttle value to an integer.", obj=task._ds, orig_exc=e)
+
+                rewind_point = len(self._workers)
+                if throttle > 0 and self.ALLOW_BASE_THROTTLING:
```

Comment:
I'm not sure it really matters, but I'd put `self.ALLOW_BASE_THROTTLING` first.

Reply:
Yes not sure it matters, in most cases throttle will most likely not be set so if there's any kind of perf hit there (like 100k iterations difference) I think it would be minimal.